### PR TITLE
Correctly print global augmentations

### LINKED ISSUE #18657: Synthesized global augmentations not printed correctly
```ts
            printsCorrectly("emptyGlobalAugmentation", {}, printer => printer.printNode(
                EmitHint.Unspecified,                                                   
                createModuleDeclaration(                                                
                    /*decorators*/ undefined,                                           
                    /*modifiers*/ [createToken(SyntaxKind.DeclareKeyword)],             
                    createIdentifier("global"),                                         
                    createModuleBlock(emptyArray),                                      
                    NodeFlags.GlobalAugmentation),                                      
                createSourceFile("source.ts", "", ScriptTarget.ES2015)                  
            ));                                                                         
```

Expected:

```ts
declare global {}
```

Actual:

```ts
declare module global
```

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
Fixes #18657

### CODE REVIEW

File: src/compiler/emitter.ts [hunk 1]
```diff
@@ -1862,8 +1862,19 @@ namespace ts {
         }
 
         function emitModuleDeclaration(node: ModuleDeclaration) {
-            emitModifiers(node, node.modifiers);
-            write(node.flags & NodeFlags.Namespace ? "namespace " : "module ");
+            if (node.flags & NodeFlags.GlobalAugmentation) {
+                if (!hasModifier(node, ModifierFlags.Ambient)) {
+                    // Always emit a 'declare' keyword in case it wasn't provided by a factory function call.
```
Comment (DanielRosenwasser) [inline src/compiler/emitter.ts]: Maybe we should just throw an error when factory functions get invalid data? What's the protocol for something like this @rbuckton ?

File: tests/baselines/reference/printerApi/printsNodeCorrectly.emptyGlobalAugmentation.js [hunk 2]
```diff
@@ -0,0 +1 @@
+declare global { }
```
Comment (sandersn) [inline tests/baselines/reference/printerApi/printsNodeCorrectly.emptyGlobalAugmentation.js]: why does this end up in a .js? I expected a .d.ts or something.

File: src/compiler/emitter.ts (hunk 1 above)
Comment (sandersn) [reply src/compiler/emitter.ts]: I think throwing an error is better, or changing the type to require the declare keyword.

Comment (mhegazy) [reply src/compiler/emitter.ts]: we do not do this in other places, why not just emit modifiers?

File: tests/baselines/reference/printerApi/printsNodeCorrectly.emptyGlobalAugmentation.js (hunk 2 above)
Comment (mhegazy) [reply tests/baselines/reference/printerApi/printsNodeCorrectly.emptyGlobalAugmentation.js]: the printer does not care what it prints to. the transforms are the one that do.

File: src/compiler/emitter.ts (hunk 1 above)
Comment (rbuckton) [reply src/compiler/emitter.ts]: `declare` isn't always legal, such as when meeting a global augmentation in a module augmentation:
```ts
// assuming there is a "foo" module...
declare module "foo" {
  global {
  }
}
```

Comment (DanielRosenwasser) [reply src/compiler/emitter.ts]: That is a good point, and @rbuckton I didn't realize that was even legal.

**Pull request merged**